client/components/embed-container/index.jsx

```diff
@@ -236,6 +237,45 @@ function embedStory( domNode ) {
 	}
 }
 
+function embedTiledGallery( domNode ) {
+	debug( 'processing tiled gallery for', domNode );
+	const galleryItems = domNode.getElementsByClassName( 'tiled-gallery__item' );
+
+	if ( galleryItems && galleryItems.length ) {
+		const imageItems = Array.from( galleryItems );
+
+		// Replace the gallery with updated markup
+		createRoot( domNode ).render(
+			<div className="gallery-container">
+				{ imageItems.map( ( item ) => {
+					const itemImage = item.querySelector( 'img' );
+					const itemLink = item.querySelector( 'a' );
+
+					const imgProps = {
+						id: itemImage?.id || undefined,
+						className: itemImage?.className || undefined,
+						src: itemImage?.src || undefined,
+					};
+
+					return (
+						<figure className="gallery-item">
+							<div className="gallery-item-wrapper">
+								{ itemLink?.href ? (
+									<a href={ itemLink.href }>
+										<img alt={ itemImage?.alt || '' } { ...imgProps } />
+									</a>
+								) : (
+									<img alt={ itemImage?.alt || '' } { ...imgProps } />
+								) }
```

Comment:
Nit: It feels like we may be asking for trouble in the long run by creating the `img` in two locations like this. Might it be worth setting the value into a constant higher up so we only set up the node in one place?
```suggestion
					const imageElement = (
						<img
							alt={ itemImage.alt || '' }
							id={ itemImage?.id || undefined }
							className={ itemImage?.className || undefined }
							src={ itemImage?.src || undefined }
							srcSet= { itemImage?.srcset || undefined }
						/>
					);

					return (
						<figure className="gallery-item">
							<div className="gallery-item-wrapper">
								{ itemLink?.href ? (
									<a href={ itemLink.href }>
										{ imageElement }
									</a>
								) : (
									{ imageElement }
								) }
```